packages/client/src/declarations/index.ts

```diff
@@ -1121,14 +1123,25 @@ function getAttachmentUrls(queryResultData: Query) {
 }
 
 function getSignatureUrls(queryResultData: Query) {
-  const registration =
-    queryResultData.fetchBirthRegistration?.registration ||
-    queryResultData.fetchDeathRegistration?.registration ||
-    queryResultData.fetchMarriageRegistration?.registration
-
-  return SIGNATURE_KEYS.map(
-    (propertyKey) => registration?.[propertyKey]
-  ).filter((maybeUrl): maybeUrl is string => Boolean(maybeUrl))
+  const data =
+    queryResultData.fetchBirthRegistration ||
+    queryResultData.fetchDeathRegistration ||
+    queryResultData.fetchMarriageRegistration
+
+  if (!data) return []
+  const { registration, history } = data
+
+  const registrarSignatures =
+    history
+      ?.map((entry) => entry?.signature?.data)
+      .filter((entry): entry is string => typeof entry === 'string') || []
```

Comment:
Would `Boolean(entry)` be a better choice here perhaps? As it would also remove "empty strings".